tests/functional/openstackoperator_controller_test.go

```diff
@@ -197,6 +198,8 @@ var _ = Describe("OpenStackOperator controller", func() {
 					},
 				},
 			}
+			// TODO: had to disable tls to allow control plane status to update
+			spec["tls"] = map[string]interface{}{}
```

Comment:
@stuggi any idea why I have to disable tls to allow the CP to get to Ready in functional tests? I see errors in the log about expires field being empty and not conforming to date-time format when the reconcile loop attempts to update Status. (and then the test case here fails because Ready states never get updated on CP).

Reply:
I think we have a bug here https://github.com/openstack-k8s-operators/openstack-operator/blob/00c81bcd4ba46e1fcb12df2a62a22eb166ea7eb7/pkg/openstack/ca.go#L138-L143

The TLSCAStatus.Expires fields expects a data-time but the above code does not initialize it if there is no certs, so it ends up being initialized to the go empty value `""`. 

I put an extra log there and I see:
```
  2024-02-09T09:55:30.298+0100	INFO	!!! Setting endpoint status	{"controller": "openstackcontrolplane", "controllerGroup": "core.openstack.org", "controllerKind": "OpenStackControlPlane", "OpenStackControlPlane": {"name":"3d79d521-10c2-4707-8462-a","namespace":"ee00d7df-3130-4715-b928-58d93c968689"}, "namespace": "ee00d7df-3130-4715-b928-58d93c968689", "name": "3d79d521-10c2-4707-8462-a", "reconcileID": "d36a4742-3150-42f4-acfd-da70f7be3dcd", "endpoint": "public", "status": {"name":"rootca-public","expires":""}, "len(caOnlyBundle.certs)": 0}
```

Reply:
In the test the PEM data is "test" (in base64) and that result in an empty bundle due to breaking here
https://github.com/openstack-k8s-operators/openstack-operator/blob/00c81bcd4ba46e1fcb12df2a62a22eb166ea7eb7/pkg/openstack/ca.go#L397-L402

Reply:
so we either need a bit better test data with certs in it or we need to make the expires field to allow being empty (maybe a pointer type allow nil without forcing the date-time validation, I don't know)

Reply:
For this test case and this PR I'm OK to keep the TLS disabled.

Reply:
we discussed in our tls meeting a change we want to make to the openstack-operator for the CA handling. I will rework on this in the next week or the following. I'll then also add more tests. for now we could disable it as in this pr